src/app/Shared/Services/Login.service.tsx

```diff
@@ -182,12 +196,30 @@ export class LoginService {
     this.username.next('');
     this.logout.next();
     this.sessionState.next(SessionState.NO_USER_SESSION);
+
+    // TODO implement a logout call that will delete the access_token
+    // cached in the oauth server, thus redirecting users to the login page
+    this.queryAuthMethod();
   }
 
   setSessionState(state: SessionState): void {
     this.sessionState.next(state);
   }
 
+  private getTokenFromUrlFragment(): string {
+    var matches = location.hash.match(new RegExp('access_token'+'=([^&]*)'));
```

Comment:
I think the regular expression here is worth extracting out to a constant. Also, this is basically just a style thing, but `new RegExp('access_token'+'=([^&]*)')` could simply be `/access_token'+'=([^&]*)/` as a regex literal.

Should the regex use anchors, since it's matching against the browser's URL location hash string already, which should be pretty well-defined/constrained?

Reply:
Ah, or the hash contains both the `access_token` and the `token_type`, so perhaps anchoring doesn't make sense. Disregard that part.